lib/foreman/renderer.rb

```diff
@@ -93,9 +93,11 @@ def indent(count)
       prefix + text.gsub(/\n/, "\n#{prefix}")
     end
 
+    # accepts either template object or plain string
     def unattended_render(template, template_name = nil)
-      content = template.respond_to?(:template) ? template.template : template
       template_name ||= template.respond_to?(:name) ? template.name : 'Unnamed'
+      raise ::Foreman::Exception.new(N_("Template '%s' is either missing or has invalid taxonomy"), template_name) if template.nil?
```

Comment:
Avoid using "taxonomy" in user shown strings if you can please, use "organization or location".